agent/agent-bootstrap/src/main/java/com/microsoft/applicationinsights/agent/bootstrap/diagnostics/DiagnosticsHelper.java

```diff
@@ -46,47 +44,40 @@ private DiagnosticsHelper() { }
 
     public static void setAgentJarFile(Path agentPath) {
         if (Files.exists(agentPath.resolveSibling("appsvc.codeless"))) {
-            appServiceCodeless = true;
+            appSvcAttachForLoggingPurposes = true;
+            if ("java".equals(System.getenv("FUNCTIONS_WORKER_RUNTIME"))) {
```

Comment:
👍 for the dedicated plan, azure functions is using app service's prefix. this will fix it :)